File: Bisiesto2Application/app/src/main/java/com/example/bisiesto2application/MainActivity.java
package com.example.bisiesto2application;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.Toast;

import com.example.bisiesto2application.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        verificar();
    }

    private void verificar() {

        binding.btnVerificar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (!TextUtils.isEmpty(binding.etIngresarAnio.getText().toString())){
                    Intent anio = new Intent(view.getContext(), DatosAnioActivity.class);
                    anio.putExtra("anio",binding.etIngresarAnio.getText().toString());
                    startActivity(anio);
                }else{
                    Toast.makeText(view.getContext(), "Ingrese un año", Toast.LENGTH_SHORT).show();
                }
            }
        });

        binding.btnSalir.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                System.exit(0);
            }
        });

    }
}

File: Prueba2406/app/src/main/java/com/example/prueba2406/MainActivity.kt
package com.example.prueba2406

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.text.TextUtils
import android.widget.Toast
import com.example.prueba2406.Operaciones.Operaciones1
import com.example.prueba2406.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        //setContentView(R.layout.activity_main)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        var op1 = Operaciones1()

        fun metodo(){
            var r = ""
            if(binding.rbSuma.isChecked){
                r = "+"
            }
            if(binding.rbResta.isChecked){
                r = "-"
            }
            if (binding.rbDivi.isChecked){
                r = "/"
            }
            if(binding.rbMulti.isChecked){
                r = "x"
            }
            op1.icono = r
        }

        fun setVaribles():Boolean{
            var flag = false
            if((!(TextUtils.isEmpty(binding.etNumero1.text.toString())))&&(!(TextUtils.isEmpty(binding.etNumero2.text.toString())))){
                flag = true
                op1.n1 = binding.etNumero1.text.toString().toDouble()
                op1.n2 = binding.etNumero2.text.toString().toDouble()
            }
            return flag
        }


        binding.btnAcciones.setOnClickListener {
            if(setVaribles()){
                metodo()
                binding.tvRespueta.text = op1.acciones()
            }else{
                Toast.makeText(this, "Ingrese todos los valores",Toast.LENGTH_SHORT)
            }
        }

        binding.btnSalir.setOnClickListener {
            finish()
        }

    }
}